### PR TITLE
Add "security" chapter to handbook

### PR DESCRIPTION
**Pull request by david-a-wheeler**
Add a chapter to explain how to write more secure software
using Stimulus.  This information isn't obvious, it's important,
and it's pretty straightforward.

This information was provided in part by Sam Stephenson
on Tue, 13 Mar 2018 16:35:18 +0000. See
https://discourse.stimulusjs.org/t/security-csp-and-stimulus/171/2

Signed-off-by: David A. Wheeler <dwheeler@dwheeler.com>

### CODE REVIEW

File: handbook/06_security.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+In any web application you should set your CSP to be restrictive as possible, regardless of whether or not it uses Stimulus.  In particular, all your JavaScript code should be in *separate* files, and the CSP should be set so that scripts are only accepted from valid sources (never from the HTML file). That way, if a value is not escaped properly (a common problem), an attacker still can't easily turn it into an XSS attack (because scripts are not run directly if they are in the HTML).
+
+CSP includes various work-arounds if you must embed scripts in the HTML, but it is better to avoid the work-arounds.  The work-arounds often make the code more complicated to maintain, and aren't as well supported.
+
+For more information, see the [OWASP CSP Cheat Sheet](https://www.owasp.org/index.php/Content_Security_Policy_Cheat_Sheet).
+
+## Stimulus does not just execute what's in the HTML
+
+CSP is valuable when using Stimulus, just like CSP is useful in other frameworks, because Stimulus does not just execute ("eval") code in the HTML.  Indeed, Stimulus does not use `eval` all.
```
Comment (merwok) [inline handbook/06_security.md]: typo: at all

Comment (david-a-wheeler) [reply handbook/06_security.md]: Good catch, will fix.

File: handbook/06_security.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+First, it's important to understand that Stimulus will *only* call classes you have defined and marked as controllers.  All controller classes must be registered with corresponding identifiers in your Stimulus `Application`.
+
+Thus, if you're trying to counter an attacker who might be able to inject something into your HTML, just ensure that none of the methods on any of your controllers perform sensitive operations without appropriate safeguards.  In these cases all controller methods should treat their inputs with suspicion, and limit the effects they can cause.
+
+## Wrap-Up and Next Steps
+
+In this chapter we've seen how to maintain security.
+
+Next we'll see how to install and configure Stimulus in your own application.
```
Comment (merwok) [inline handbook/06_security.md]: This line is also in Chapter 05

Comment (david-a-wheeler) [reply handbook/06_security.md]: Ah, chapter 05 needs to change to introduce the new chapter 06.  I'll post an update soon.

**Comment by david-a-wheeler** [comment]
I've posted an update on this branch that fixes the issues noted by Éric Araujo (@merwok).

**Comment by david-a-wheeler** [comment]
As always, let me know of any other issues. I *think* it's okay now (famous last words).

**Comment by sstephenson** [comment]
David, thanks again for your work here. I apologize for letting this pull request sit so long. Sometimes I have difficulty saying "no" to good contributions when they don't fit perfectly with a project's goals or opinions.

Ultimately, I'd rather maintain a narrower scope on security recommendations. I made an attempt at that in #184 based on our initial Q&A conversation. I'd love to hear your thoughts on that PR if you have the time.

**Pull request closed without merging**